### ISSUE TITLE
TypeError in RTF format

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#755]
(Original issue 1048 created by ymhuang0808 on 2014-10-28T02:46:34.327421+00:00)

An error occurs in PHP RTF format.

The error message is below:
*** Error while highlighting:
TypeError: ord() expected a character, but string of length 3 found
   (file "/Library/Python/2.7/site-packages/pygments/formatters/rtf.py", line 77, in _escape_text)
                {\cf2 'description'} {\cf4 =>} {\cf2

Do I show the target file?

**Comment by Anteru** [comment]
(Original issue 1048 last updated on 2014-10-29T06:55:34.324090+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by ymhuang0808 on 2014-10-28T03:02:09.474466+00:00)

Hi, 
I find the problem. Because the line contains Traditional Chinese, I change the Traditional Chinese into English. Does Pygments support Traditional Chinese?

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-28T08:35:10.521629+00:00)

In principle all Unicode is supported, but this may be an issue with using a narrow-unicode build of Python and characters outside the BMP.

Can you attach a minimal example file that shows this error, and the command line you used for `pygmentize`? Also, which Pygments version is this?

**Comment by Anteru** [comment]
(Original comment by ymhuang0808 on 2014-10-28T09:07:56.035110+00:00)

Pygments version is 1.6 . 

The command is:

```
#!shell

$ pygmentize -f rtf EventTest.php
```


My example file:

```
#!php

<?php

class EvnetTest extends \PHPUnit_Framework_TestCase
{
    public function testSetAndGetEvent()
    {
        $event = new \PHPUnitEvent\Event();
        $eventArray1 = array(
            [
                'name' => 'PHPUnit', 
                'start_date' => '2014-10-27 09:00', 
                'end_date' => '2014-10-27 12:00',
                'description' => 'PHPUnit 教學 XD',
            ]
        );
        $event->setEvents($eventArray1);
        $this->assertCount(1, $event->getEvents());
    }
}
```

The error occurs in line 13.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-28T09:12:28.399900+00:00)

Please attach the file instead, when copying from the web page I can't be sure the encoding is preserved.

**Comment by Anteru** [comment]
(Original comment by ymhuang0808 on 2014-10-28T09:14:01.086959+00:00)

The file link in Github: [https://raw.githubusercontent.com/ymhuang0808/PHPUnit-Event-example/996094ed369efcf7a4c33b20648c714174a23274/tests/EventTest.php](https://raw.githubusercontent.com/ymhuang0808/PHPUnit-Event-example/996094ed369efcf7a4c33b20648c714174a23274/tests/EventTest.php)

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-28T16:36:30.707942+00:00)

Thanks. I'll have a look.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-29T06:50:49.250474+00:00)

OK, this is likely already fixed in 2.0rc1, parts of the RTF handling code have been rewritten there. Thanks for the report!

**Comment by Anteru** [comment]
(Original comment by ymhuang0808 on 2014-10-29T06:55:34.322280+00:00)

Thanks for your help :)
